src/common/test/test_common.h

```diff
@@ -16,14 +18,17 @@ extern int usleep(useconds_t usec);
 #endif
 
 /* I/O helper methods to retry binding to sockets. */
-static inline UT_string *bind_ipc_sock_retry(const char *socket_name_format,
-                                             int *fd) {
-  UT_string *socket_name = NULL;
+static inline std::string bind_ipc_sock_retry(const char *socket_name_format,
+                                              int *fd) {
+  std::string socket_name;
   for (int num_retries = 0; num_retries < 5; ++num_retries) {
     LOG_INFO("trying to find plasma socket (attempt %d)", num_retries);
-    utstring_renew(socket_name);
-    utstring_printf(socket_name, socket_name_format, rand());
-    *fd = bind_ipc_sock(utstring_body(socket_name), true);
+    size_t size = std::snprintf(nullptr, 0, socket_name_format, rand()) + 1;
```

Comment:
It's ok if you use std::snprintf here unless it is easy to port it to c++ stringstreams instead (this code will be ported to gtest at some point and we can do the conversion then.

Reply:
C++ stringstreams are more difficult here since `bind_ipc_sock_retry` accepts a format string argument. I'll leave it as `std::snprintf` for now.

Reply:
yes, sounds good!